mmv1/templates/terraform/resource.go.tmpl

```diff
@@ -288,13 +288,29 @@ func resource{{ $.ResourceName -}}Create(d *schema.ResourceData, meta interface{
 {{- end}}
         return fmt.Errorf("Error creating {{ $.Name -}}: %s", err)
     }
-{{- /* # Set resource properties from create API response (unless it returns an Operation) */}}
-{{- if not (and $.GetAsync ($.GetAsync.IsA "OpAsync")) }}
+{{- /* Set output-only resource properties from create API response (as long as Create doesn't use an async operation) */}}
+{{- /* This is necessary so that the ID is built correctly. */}}
+{{- if or (not $.GetAsync) (not (and ($.GetAsync.IsA "OpAsync") ($.GetAsync.Allow "Create"))) }}
 {{- range $prop := $.GettableProperties }}
-{{-  if and ($.IsInIdentity $prop) $prop.Output }}
+{{-  if $.InIdFormat $prop }}
+{{-  if eq $prop.CustomFlatten "templates/terraform/custom_flatten/id_from_name.tmpl" }}
```

Comment:
This works... but if we are directly building logic around a specific custom flatten, maybe the flattener is leaving the realm of custom code? Should `id_from_name` just be a field-level toggle and we move `id_from_name.tmpl` to `mmv1/terraform/templates`?

Then we can add it to https://googlecloudplatform.github.io/magic-modules/reference/field/ and explain when to use it?

Reply:
If there's agreement on this, but it's out of scope for this change, I'm okay with that. Especially since if we switch it to a field-level toggle, there should be no diffs generated from this PR

Reply:
yes, I think there's probably some improvement to make here (such as making id_from_name a field-level toggle or similar) but I didn't want to get into it as part of this PR.